packages/client-assertion-validation/src/validation.ts

```diff
@@ -184,11 +186,14 @@ export const verifyClientAssertion = (
 
 export const verifyClientAssertionSignature = async (
   clientAssertionJws: string,
-  key: Key
+  key:
+    | TokenGenerationStatesClientPurposeEntry
+    | TokenGenerationStatesClientEntry,
+  algorithm: string
```

Comment:
just to make it clear which value is required

```suggestion
  clientAssertionAlgorithm: string
```

Reply:
Renamed